### PR TITLE
r/aws_fsx_lustre_file_system - support modification of per_unit_storage_throughput parameter

### LINKED ISSUE #34862: [Enhancement]: Support for resizing throughput requirements for FSx Lustre
### Description

Prior to 11/2023, FSx for Lustre did not allow end users to change or modify the throughput requirements for FSxL filesystems; the filesystem had to be destroyed and recreated. That changed based on [this](https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-fsx-lustre-adjust-throughput-tier-file-systems/) announcement. Enhancement for support for modifications to the `per_unit_storage_throughput` key without recreation and destruction of the filesystem.

### Affected Resource(s) and/or Data Source(s)

"aws_fsx_lustre_file_system"

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/fsx_lustre_file_system.html#per_unit_storage_throughput

### Potential Terraform Configuration

```terraform
No change, aside from ForceNew = false in https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/fsx/lustre_file_system.go.
```

### References

https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-fsx-lustre-adjust-throughput-tier-file-systems/

### Would you like to implement a fix?

No

### PR DESCRIPTION
**Pull request by sasidhar-aws**
### Description
aws_fsx_lustre_file_system supports modification of per_unit_storage_throughput parameter without recreation and destruction of the filesystem.

### Relations
Closes #34862

### References
https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-fsx-lustre-adjust-throughput-tier-file-systems/

### Output from Acceptance Testing

```console
make testacc TESTS=TestAccFSxLustreFileSystem_deploymentTypePersistent PKG=fsx
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/fsx/... -v -count 1 -parallel 20 -run='TestAccFSxLustreFileSystem_deploymentTypePersistent'  -timeout 360m
=== RUN   TestAccFSxLustreFileSystem_deploymentTypePersistent1
=== PAUSE TestAccFSxLustreFileSystem_deploymentTypePersistent1
=== RUN   TestAccFSxLustreFileSystem_deploymentTypePersistent1_perunitstoragethroughput
=== PAUSE TestAccFSxLustreFileSystem_deploymentTypePersistent1_perunitstoragethroughput
=== RUN   TestAccFSxLustreFileSystem_deploymentTypePersistent2
=== PAUSE TestAccFSxLustreFileSystem_deploymentTypePersistent2
=== RUN   TestAccFSxLustreFileSystem_deploymentTypePersistent2_perunitstoragethroughput
=== PAUSE TestAccFSxLustreFileSystem_deploymentTypePersistent2_perunitstoragethroughput
=== CONT  TestAccFSxLustreFileSystem_deploymentTypePersistent1
=== CONT  TestAccFSxLustreFileSystem_deploymentTypePersistent2
=== CONT  TestAccFSxLustreFileSystem_deploymentTypePersistent2_perunitstoragethroughput
=== CONT  TestAccFSxLustreFileSystem_deploymentTypePersistent1_perunitstoragethroughput
--- PASS: TestAccFSxLustreFileSystem_deploymentTypePersistent2 (613.61s)
--- PASS: TestAccFSxLustreFileSystem_deploymentTypePersistent1 (674.79s)
--- PASS: TestAccFSxLustreFileSystem_deploymentTypePersistent2_perunitstoragethroughput (925.38s)
--- PASS: TestAccFSxLustreFileSystem_deploymentTypePersistent1_perunitstoragethroughput (958.70s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/fsx	968.041s
```

### CODE REVIEW

File: internal/service/fsx/lustre_file_system.go [hunk 1]
```diff
@@ -510,14 +509,18 @@ func resourceLustreFileSystemUpdate(ctx context.Context, d *schema.ResourceData,
 			input.LustreConfiguration.DailyAutomaticBackupStartTime = aws.String(d.Get("daily_automatic_backup_start_time").(string))
 		}
 
-		if v, ok := d.GetOk("data_compression_type"); ok {
```
Comment (AdamTylerLynch) [inline internal/service/fsx/lustre_file_system.go:513]: Can you elaborate on why you modified this attributes behavior?

Comment (sasidhar-aws) [reply internal/service/fsx/lustre_file_system.go:513]: Couple of reasons:
1. To keep logic consistent, where we check if value is changed while updating any argument.
2. aws fsx update-file-system API is not working in below scenario: 
           when we are passing 2 parameters to LustreConfiguration and one of the parameter value do not have any difference from old value. UpdateFileSystem is working for LustreConfiguration only when we pass all parameters which needs to be updated with new value.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccFSxLustreFileSystem_dataCompression\|TestAccFSxLustreFileSystem_basic\|TestAccFSxLustreFileSystem_deploymentTypePersistent1_per\|TestAccFSxLustreFileSystem_deploymentTypePersistent2_per' PKG=fsx ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/fsx/... -v -count 1 -parallel 2  -run=TestAccFSxLustreFileSystem_dataCompression\|TestAccFSxLustreFileSystem_basic\|TestAccFSxLustreFileSystem_deploymentTypePersistent1_per\|TestAccFSxLustreFileSystem_deploymentTypePersistent2_per -timeout 360m
=== RUN   TestAccFSxLustreFileSystem_basic
=== PAUSE TestAccFSxLustreFileSystem_basic
=== RUN   TestAccFSxLustreFileSystem_dataCompression
=== PAUSE TestAccFSxLustreFileSystem_dataCompression
=== RUN   TestAccFSxLustreFileSystem_deploymentTypePersistent1_perUnitStorageThroughput
=== PAUSE TestAccFSxLustreFileSystem_deploymentTypePersistent1_perUnitStorageThroughput
=== RUN   TestAccFSxLustreFileSystem_deploymentTypePersistent2_perUnitStorageThroughput
=== PAUSE TestAccFSxLustreFileSystem_deploymentTypePersistent2_perUnitStorageThroughput
=== CONT  TestAccFSxLustreFileSystem_basic
=== CONT  TestAccFSxLustreFileSystem_deploymentTypePersistent1_perUnitStorageThroughput
--- PASS: TestAccFSxLustreFileSystem_basic (607.46s)
=== CONT  TestAccFSxLustreFileSystem_dataCompression
--- PASS: TestAccFSxLustreFileSystem_deploymentTypePersistent1_perUnitStorageThroughput (982.52s)
=== CONT  TestAccFSxLustreFileSystem_deploymentTypePersistent2_perUnitStorageThroughput
--- PASS: TestAccFSxLustreFileSystem_dataCompression (737.10s)
--- PASS: TestAccFSxLustreFileSystem_deploymentTypePersistent2_perUnitStorageThroughput (899.03s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/fsx	1887.529s
```

**Comment by ewbankkit** [comment]
@sasidhar-aws Thanks for the contribution :tada: :clap:.

**Pull request merged**